qpc/insights/tests_insights_configure.py

```diff
@@ -0,0 +1,86 @@
+"""Test the CLI module."""
+
+import sys
+import unittest
+from io import StringIO
+
+from qpc.cli import CLI
+from qpc.tests_utilities import HushUpStderr, redirect_stdout
+from qpc.utils import DEFAULT_INSIGHTS_CONFIG, read_insights_config,\
+    write_insights_config, write_server_config
+
+
+class InsightsConfigureTests(unittest.TestCase):
+    """Class for testing insights host configuration."""
+
+    def setUp(self):
+        """Create test setup."""
+        write_server_config({'host': '127.0.0.1', 'port': 8000, 'use_http': True,
+                             'require_token': False})
+        # Temporarily disable stderr for these tests, CLI errors clutter up
+        self.orig_stderr = sys.stderr
+        sys.stderr = HushUpStderr()
+
+    def tearDown(self):
+        """Remove test case setup."""
+        # Restore stderr
+        sys.stderr = self.orig_stderr
+
+    def test_insights_config_bad_port(self):
+        """Testing insights configure when receiving bad port."""
+        with self.assertRaises(SystemExit):
+            sys.argv = ['/bin/qpc', 'insights', 'config',
+                        '--port', 'abc']
+            CLI().main()
+
+    def test_success_default_config_no_args(self):
+        """Testing if method returns default config dict when no arguments."""
+        config_out = StringIO()
+        sys.argv = ['/bin/qpc', 'insights', 'config']
+        with redirect_stdout(config_out):
+            CLI().main()
+        config = read_insights_config()
+        self.assertEqual(config['host'], DEFAULT_INSIGHTS_CONFIG['host'])
+        self.assertEqual(config['port'], DEFAULT_INSIGHTS_CONFIG['port'])
+        self.assertEqual(config['use_http'], DEFAULT_INSIGHTS_CONFIG['use_http'])
+
+    def test_success_config_insights(self):
+        """Testing insights configure green path."""
+        config_out = StringIO()
+        sys.argv = ['/bin/qpc', 'insights', 'config',
+                    '--host', 'console.insights.test', '--port', '200', '--use-http']
+        with redirect_stdout(config_out):
+            CLI().main()
+            config = read_insights_config()
+            self.assertEqual(config['host'], 'console.insights.test')
+            self.assertEqual(config['port'], 200)
+            self.assertEqual(config['use_http'], True)
```

Comment:
The assert lines can definitely be outside the with block. But it comes to my atttention this stdout redirection context (`redirect_stdout`) is not used consistently across tests in this same module.